### PR TITLE
build: to add docker/k8 setup to deriv com

### PR DESCRIPTION
**Pull request by yaswanth-deriv**
Changes:

-   Added Docker image, validated the Nginx image, pushed the image to Docker Hub, and deployed it to Kubernetes in both the staging and production environments.

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [X] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [X] Release

### CODE REVIEW

File: .github/workflows/production.yml [hunk 1]
```diff
@@ -52,7 +52,32 @@ jobs:
...
+                  docker push  ${{ secrets.DOCKHUB_ORGANISATION }}/deriv-com:latest
+                  docker push  ${{ secrets.DOCKHUB_ORGANISATION }}/deriv-com:$GIT_TAG_NAME
+
+            - name: Deploy 🚀
+              run: |
+                  export NAMESPACE="deriv-com-production"
+                  export SERVICEACCOUNT_TOKEN=${{ secrets.SERVICEACCOUNT_TOKEN }}
+                  export DOCKERHUB_ORGANISATION=${{ secrets.DOCKERHUB_ORGANISATION }}
+                  export KUBE_SERVER=${{ secrets.KUBE_SERVER }}
+                  export CA=${{ secrets.CA }}
```
Comment (ali-hosseini-deriv) [inline .github/workflows/production.yml:76]: CA is not needed @yaswanth-deriv

File: .github/workflows/staging.yml [hunk 2]
```diff
@@ -49,7 +49,32 @@ jobs:
...
+                docker push  ${{ secrets.DOCKHUB_ORGANISATION }}/deriv-com:latest-staging
+                docker push  ${{ secrets.DOCKHUB_ORGANISATION }}/deriv-com:$GITHUB_SHA
+
+            - name: Deploy 🚀
+              run: |
+                export NAMESPACE="deriv-com-staging"
+                export SERVICEACCOUNT_TOKEN=${{ secrets.SERVICEACCOUNT_TOKEN }}
+                export DOCKERHUB_ORGANISATION=${{ secrets.DOCKERHUB_ORGANISATION }}
+                export KUBE_SERVER=${{ secrets.KUBE_SERVER }}
+                export CA=${{ secrets.CA }}
```
Comment (ali-hosseini-deriv) [inline .github/workflows/staging.yml:73]: not needed

**Review by ali-hosseini-deriv — COMMENTED** [review]
left two comments

**Review by ali-hosseini-deriv — APPROVED** [review]
LGTM

**Review by prince-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]

File: .github/workflows/production.yml [hunk 3]
```diff
@@ -52,7 +52,28 @@ jobs:
...
+                  docker push  ${{ secrets.DOCKHUB_ORGANISATION }}/deriv-com:latest
+                  docker push  ${{ secrets.DOCKHUB_ORGANISATION }}/deriv-com:$GIT_TAG_NAME
+
+            - name: Deploy 🚀
+              run: |
+                  export NAMESPACE="deriv-com-production"
+                  git clone https://github.com/binary-com/devops-ci-scripts
+                  cd devops-ci-scripts/k8s-build_tools
+                  echo ${{ secrets.CA_CRT}} | base64 --decode > ca.crt
+                  ./release.sh deriv-com $GIT_TAG_NAME
```
Comment (ali-hosseini-deriv) [inline .github/workflows/production.yml:76]: @yaswanth-deriv please move them to after publish to cloudflare.
the reason is that we dont want to block the releases at least for now until it gets to a stable point.

**Review by ali-hosseini-deriv — APPROVED** [review]

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**